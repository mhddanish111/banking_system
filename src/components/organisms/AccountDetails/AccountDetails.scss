@import "../../../styles/colors.scss";
@import "../../../styles/fonts.scss";
@import "../../../styles/mixins.scss";
@import "../../../styles/spacing.scss";
.details-container {
  display: flex;
  flex-direction: column;
}

.error-message {
  color: $color-red-50;
  margin-bottom: $space-xs;
}