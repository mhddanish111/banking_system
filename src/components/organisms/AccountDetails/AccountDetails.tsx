import React from 'react';
import { get, isEmpty } from 'lodash';
import { AccountDetailsProps } from '../../../type/AccountType';
import './AccountDetails.scss';

const AccountDetails: React.FC<any> = (props: AccountDetailsProps) => {
  const { accountDetails } = props;
  const { categoryName, accountNumber, amount, limit, users, type } =
    accountDetails;
  const getAccountHolderName = () => {
    const { firstName, lastName } = get(users, ['0'], {});
    let name = '';
    if (!isEmpty(firstName) || !isEmpty(lastName)) {
      name = `${firstName} ${lastName}`;
    }
    return name;
  };
  return (
    <div className="col-xs-12 details-container">
      <div className="row account-details-box">
        <label className="col-xs-6">{type}</label>
        <div className="col-xs-6 profile-text">{categoryName}</div>
      </div>
      <div className="row account-details-box">
        <label className="col-xs-6">{`${type} Holder Name`}</label>
        <div className="col-xs-6 profile-text">{getAccountHolderName()}</div>
      </div>
      <div className="row account-details-box">
        <label className="col-xs-6">{`${type} Number`}</label>
        <div className="col-xs-6 profile-text">{accountNumber}</div>
      </div>
      <div className="row account-details-box">
        <label className="col-xs-6">{`Available ${
          type === 'Cards' ? 'Limit' : 'Amount'
        }`}</label>
        <div className="col-xs-6 profile-text">
          {type === 'Cards' ? limit : amount}
        </div>
      </div>
      {type === 'Cards' && (
        <div className="row account-details-box">
          <label className="col-xs-6">Available Amount</label>
          <div className="col-xs-6 profile-text">{amount}</div>
        </div>
      )}

      {type === 'Cards' && (
        <div className="row account-details-box">
          <label className="col-xs-6">Due Amount </label>
          <div className="col-xs-6 profile-text">{amount - limit}</div>
        </div>
      )}
    </div>
  );
};

export default AccountDetails;
