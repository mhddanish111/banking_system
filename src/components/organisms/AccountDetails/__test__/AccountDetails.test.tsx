import { shallow } from 'enzyme';
import AccountDetails from '../AccountDetails';

describe('<AccountDetails />', () => {
  const accountDetails = {
    _id: '63c1519e9919f0f41a86bd94',
    type: 'Accounts',
    accountType: 'Savings Account',
    categoryName: 'Regular Account',
    accountNumber: 123456789,
    amount: 10000,
    debitCardCSVNumber: 123,
    debitCardNumber: 12312313123,
    expiryMonth: 1,
    expiryYear: 2023,
    limit: 0,
    maxAmount: 0,
    minAmount: 0,
    charge: 0,
    users: [
      {
        _id: '63c1519e9919f0f41a86bd94',
        firstName: 'test',
        lastName: 'test',
      },
    ],
  };

  const wrapped = shallow(<AccountDetails accountDetails={accountDetails} />);
  it('should render the Title Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
