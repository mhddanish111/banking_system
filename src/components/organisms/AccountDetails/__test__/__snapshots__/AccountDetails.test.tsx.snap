// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountDetails /> should render the Title Component correctly 1`] = `
<div
  className="col-xs-12 details-container"
>
  <div
    className="row account-details-box"
  >
    <label
      className="col-xs-6"
    >
      Accounts
    </label>
    <div
      className="col-xs-6 profile-text"
    >
      Regular Account
    </div>
  </div>
  <div
    className="row account-details-box"
  >
    <label
      className="col-xs-6"
    >
      Accounts Holder Name
    </label>
    <div
      className="col-xs-6 profile-text"
    >
      test test
    </div>
  </div>
  <div
    className="row account-details-box"
  >
    <label
      className="col-xs-6"
    >
      Accounts Number
    </label>
    <div
      className="col-xs-6 profile-text"
    >
      123456789
    </div>
  </div>
  <div
    className="row account-details-box"
  >
    <label
      className="col-xs-6"
    >
      Available Amount
    </label>
    <div
      className="col-xs-6 profile-text"
    >
      10000
    </div>
  </div>
</div>
`;
